dashboard/app/controllers/script_levels_controller.rb

```diff
@@ -189,7 +189,9 @@ def stage_extras
         @section = current_user.sections[0]
         @user = @section&.students&.find_by(id: params[:user_id])
       end
-      @show_stage_extras_warning = !@section&.stage_extras
+      # This airs on the side of not showing the warning by not showing it if the script we are in
+      # is not the assigned script for the section
+      @show_stage_extras_warning = !@section&.stage_extras && @section&.script&.name == params[:script_id]
```

Comment:
It seems confusing to check the script name against the script_id?

Reply:
ya for some reason script_id here gives something like 'csd3-2019'

Reply:
the param name `script_id` is just the default you get as a result of this line of config: https://github.com/code-dot-org/code-dot-org/blob/47830740bba51f736dd92911b313e74e8bfe63f3/dashboard/config/routes.rb#L247

you could change the param name via `param: :script_name` as described here: https://guides.rubyonrails.org/routing.html#overriding-named-route-parameters

I don't think that's necessarily worth doing at this time, so for now the next-best practice is to confine the poorly named param to the controller (which you are doing), and not let the bad param name leak down into a view or model.